# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/DMABufLibWrapper.cpp
# Commit: 4c4c96ad829d
# Full Hash: 4c4c96ad829d55b88ee52a320cab335ac2b86340
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-05 16:56:59
# Regressor Bug: 1944490
# File Overlap Count: 1
# Description:
#   Bug 1944490 [Linux] Ensure DMABufFormats provides basic dmabuf formats r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D236688
# ==============================================================================

diff -r 25e327e7c9ad -r 4c4c96ad829d widget/gtk/DMABufLibWrapper.cpp
--- a/widget/gtk/DMABufLibWrapper.cpp	Tue Feb 04 20:14:58 2025 +0000
+++ b/widget/gtk/DMABufLibWrapper.cpp	Tue Feb 04 20:44:48 2025 +0000
@@ -246,9 +246,8 @@
 #ifdef MOZ_WAYLAND
 void DMABufDevice::SetModifiersToGfxVars() {
   RefPtr<DMABufFormats> formats = WaylandDisplayGet()->GetDMABufFormats();
-  if (!formats) {
-    return;
-  }
+  MOZ_RELEASE_ASSERT(formats, "Missing dmabuf formats!");
+
   if (DRMFormat* format = formats->GetFormat(GBM_FORMAT_XRGB8888)) {
     mFormatRGBX = new DRMFormat(*format);
     gfxVars::SetDMABufModifiersXRGB(*format->GetModifiers());
@@ -272,8 +271,10 @@
 RefPtr<DRMFormat> DMABufDevice::GetDRMFormat(int32_t aFOURCCFormat) {
   switch (aFOURCCFormat) {
     case GBM_FORMAT_XRGB8888:
+      MOZ_DIAGNOSTIC_ASSERT(mFormatRGBX, "Missing RGBX dmabuf format!");
       return mFormatRGBX;
     case GBM_FORMAT_ARGB8888:
+      MOZ_DIAGNOSTIC_ASSERT(mFormatRGBA, "Missing RGBA dmabuf format!");
       return mFormatRGBA;
     default:
       gfxCriticalNoteOnce << "DMABufDevice::GetDRMFormat() unknow format: "